### ISSUE TITLE
Request body object is passed as-is

### ISSUE THREAD

**Issue by kettanaito** [github/fetch#856]
## Current behavior

When providing a non-stringified object as a request body, `whatwg-fetch` propagates that object to `XMLHttpRequest.send()` as-is.

```js
fetch('/example', { 
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  },
  body: { a: 1 }
})
```

## Expected behavior

Expected for this scenario to behave as native `window.fetch`, where such non-stringified request body would be transformed into the `"[object Object]"` string, stating that this is not a valid body per request headers.

## Justification

This raises an incongruence between the polyfill and the implementation it polyfills, producing a different behavior in a browser and NodeJS for the same request issuing code. 

- Originated from https://github.com/mswjs/msw/issues/433

## Reproduction

See this [reproduction repository](https://github.com/kettanaito/whatwg-fetch-json).

**Comment by kettanaito** [comment]
I understand this is a slightly different testing context between the two cases, but I hope it showcases the issue.

## Browser

Unlike `whatwg-fetch`, `window.fetch` in the browser doesn't rely on `XMLHttpRequest` class, so the most objective way to see what it does with the non-stringified body is to inspect the "Request payload' in the network of DevTools.

## NodeJS

In NodeJS I can see that `whatwg-fetch` dispatched `XMLHttpRequest.open` with the request body as-is. By inspecting the polyfill implementation I also conclude that there's no logic that would attempt to stringify the body:

https://github.com/github/fetch/blob/61372fdeb8e6eb59e13da265d518dca21467c1d3/fetch.js#L216-L258

**Comment by miccycn** [comment]
Same issue

**Comment by JakeChampion** [comment]
@kettanaito @miccycn, would either of you be willing to submit a pull-request which fixes this?

**Comment by kettanaito** [comment]
@JakeChampion, sure thing. I'd be glad to ship a fix.

**Comment by monstro805** [comment]
![Copy of repository-open-graph-template](https://user-images.githubusercontent.com/76422372/108236033-4e700b00-70fb-11eb-9ae3-4566526ba92d.png)

**Comment by kettanaito** [comment]
Hey, @JakeChampion. Please, would there be a chance to review my pull request (https://twitter.com/ApiMocking/status/1362084398632488961)? That'd be of huge help.

**Comment by JakeChampion** [comment]
> Hey, @JakeChampion. Please, would there be a chance to review my pull request (#881)? That'd be of huge help.

The pull-request was opened as a draft so I assumed it was not ready for a review. I've reviewed it now.  Thank you for the contribution :-)

**Comment by kettanaito** [comment]
Thank you for the review, @JakeChampion! 🎉 My bad, I haven't marked it as ready. Excited to see this merged and looking forward to the release!

**Comment by JakeChampion** [comment]
This is released in v3.6.0 -- https://github.com/github/fetch/releases/tag/v3.6.0
